src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -160,29 +168,137 @@ public virtual bool TryFindConnection(string ownerUri, out ConnectionInfo connec
                 };
             }
 
-            // Resolve if it is an existing connection
-            // Disconnect active connection if the URI is already connected
+            // return null upon success
+            return null;
```

Comment:
This code should eventually take a page from the QueryExecutionService and send Error on failure, and a Result on success. This would simplify API quite a bit, but would be a fairly substantial change. I say it's outside the scope for this change, but we should backlog it.